### PR TITLE
Add API test case for retag

### PR DESCRIPTION
**Pull request by danfengliu**
Add API test case for retag, create a user and 2 projects by this new user, push image to 1st project, and retag this image to 2nd project.

Signed-off-by: danfengliu <danfengl@vmware.com>

### CODE REVIEW

File: tests/apitests/python/test_retag.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        Test step and expected result:
+            1. Create a new user(UA);
+            2. Create a new project(PA) by user(UA);
+            3. Create a new project(PB) by user(UA);
+            4. Create a new repository(RA) in project(PA) by user(UA);
+            5. Get repository in project(PA), there should be one repository which was created by user(UA);
+            6. Get repository(RA)'s tag detail information;
+            7. Get repository(RA)'s image tag detail information;
+            8. Retag repository(RA)'s image to project(PB);
+            9. Read digest of retaged image, it must be the same with the image in repository(RA).
```
Comment (wy65701436) [inline tests/apitests/python/test_retag.py]: add pull RA from PB.

File: tests/apitests/python/test_retag.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        self.project.delete_project(TestProjects.project_dst_repo_id, **TestProjects.USER_RETAG_CLIENT)
+
+        #4. Delete user(UA).
+        self.user.delete_user(TestProjects.user_retag_id, **ADMIN_CLIENT)
+
+    def testRetag(self):
+        """
+        Test case:
+            Retag Image
+        Test step and expected result:
```
[the remark this answers is not in the document]
Comment (danfengliu) [reply tests/apitests/python/test_retag.py]: It's done as your advise.

File: tests/apitests/python/test_retag.py (hunk 1 above)
Comment (danfengliu) [reply tests/apitests/python/test_retag.py]: It's done as your advise.

File: tests/apitests/python/library/project.py [hunk 3]
```diff
@@ -131,6 +137,15 @@ def get_project_member(self, project_id, member_id, expect_status_code = 200, ex
         base._assert_status_code(200, status_code)
         return data
 
+    def get_project_member_id(self, project_id, member_user_name, **kwargs):
+        members = self.get_project_members(project_id, **kwargs)
+        print members
```
Comment (wy65701436) [inline tests/apitests/python/library/project.py]: remove the print

File: tests/apitests/python/library/project.py [hunk 4]
```diff
@@ -15,6 +15,12 @@ def is_member_exist_in_project(members, member_user_name, expected_member_role_i
                 return True
     return result
 
+def get_member_id_in_project(members, member_user_name):
```
Comment (wy65701436) [inline tests/apitests/python/library/project.py]: why do we need a method here?

File: tests/apitests/python/test_retag.py [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+        self.project.update_project_member_role(TestProjects.project_dst_repo_id, retag_member_id, 1, **ADMIN_CLIENT)
+
+        #10. Retag image in project(PA) to project(PB), it should be successful;
+        self.repo.retag_image(TestProjects.dst_repo_name, dst_tag_name, TestProjects.src_repo_name+":"+src_tag_data.digest, **TestProjects.USER_RETAG_CLIENT)
+
+        #11. Get repository(RB)'s image tag detail information;
+        dst_tag_data = self.repo.get_tag(TestProjects.dst_repo_name, dst_tag_name, **TestProjects.USER_RETAG_CLIENT)
+
+        #12. Read digest of retaged image, it must be the same with the image in repository(RA);
+        _assert_status_code(src_tag_data.digest, dst_tag_data.digest)
```
Comment (wy65701436) [inline tests/apitests/python/test_retag.py]: _assert_status_code == the name is confusing, acturally it's not for code assertion

File: tests/apitests/python/library/project.py (hunk 4 above)
Comment (wy65701436) [reply tests/apitests/python/library/project.py]: the name should be get_member_id_by_name

File: tests/apitests/python/library/docker_api.py [hunk 6]
```diff
@@ -28,6 +28,8 @@ def docker_image_pull(self, image, tag = None, expected_error_message = None):
             expected_error_message = None
         caught_err = False
         ret = ""
+        print "image:",image
```
Comment (wy65701436) [inline tests/apitests/python/library/docker_api.py]: remove the print

File: tests/apitests/python/test_retag.py (hunk 5 above)
Comment (wy65701436) [reply tests/apitests/python/test_retag.py]: can we leverage the assertion in python?

**Pull request merged**